cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/FileSetProperties.java

```diff
@@ -50,18 +55,133 @@
    */
   public static final String OUTPUT_PROPERTIES_PREFIX = "output.properties.";
 
+  /**
+   * Whether this dataset should be enabled for explore.
+   */
+  public static final String PROPERTY_EXPLORE_ENABLED = "explore.enabled";
```

Comment:
maybe rename to explore.enable.on.create or something with that meaning.  Otherwise it sounds like a status that reflects whether or not explore is enabled, which is independent of this property.